
# HG changeset patch
# User Timo Sirainen <tss@iki.fi>
# Date 1416386238 -32400
# Node ID 31262a892ba7302bfea6e70e17d3acd468736d70
# Parent  8973329d1ceb949247dd7b852d707215bfe6af17
mbox: istream_raw_mbox_seek() needs to reset all corruption errors.
We now reset all errors, but it doesn't really matter.

diff -r 8973329d1ceb -r 31262a892ba7 src/lib-storage/index/mbox/istream-raw-mbox.c
--- a/src/lib-storage/index/mbox/istream-raw-mbox.c	Sat Nov 15 03:26:16 2014 +0200
+++ b/src/lib-storage/index/mbox/istream-raw-mbox.c	Wed Nov 19 17:37:18 2014 +0900
@@ -688,6 +688,9 @@
 
 	i_assert(rstream->locked);
 
+	/* reset any (corruption) errors */
+	stream->stream_errno = 0;
+	i_free_and_null(stream->real_stream->iostream.error);
 	rstream->corrupted = FALSE;
 	rstream->eof = FALSE;
 	rstream->istream.istream.eof = FALSE;

